app/src/main/java/com/realityexpander/tasky/agenda_feature/data/repositories/eventRepository/eventRepositoryImpls/EventRepositoryImpl.kt

```diff
@@ -0,0 +1,89 @@
+package com.realityexpander.tasky.agenda_feature.data.repositories.eventRepository.eventRepositoryImpls
+
+import com.realityexpander.tasky.agenda_feature.data.common.convertersDTOEntityDomain.toDTO
+import com.realityexpander.tasky.agenda_feature.data.common.convertersDTOEntityDomain.toDomain
+import com.realityexpander.tasky.agenda_feature.data.common.convertersDTOEntityDomain.toEntity
+import com.realityexpander.tasky.agenda_feature.data.repositories.eventRepository.local.eventDao.IEventDao
+import com.realityexpander.tasky.agenda_feature.data.repositories.eventRepository.local.eventDao.eventDaoImpls.EventDaoFakeImpl
+import com.realityexpander.tasky.agenda_feature.data.repositories.eventRepository.remote.eventApi.IEventApi
+import com.realityexpander.tasky.agenda_feature.data.repositories.eventRepository.remote.eventApi.eventApiImpls.EventApiFakeImpl
+import com.realityexpander.tasky.agenda_feature.domain.AgendaItem
+import com.realityexpander.tasky.agenda_feature.domain.IEventRepository
+import com.realityexpander.tasky.agenda_feature.util.EventId
+import java.time.ZonedDateTime
+
+class EventRepositoryImpl(
+    private val eventDao: IEventDao = EventDaoFakeImpl(),
+    private val eventApi: IEventApi = EventApiFakeImpl(),
+) : IEventRepository {
+
+    override suspend fun createEvent(event: AgendaItem.Event): Boolean {
+        return try {
+            eventDao.createEvent(event.toEntity())
+            eventApi.createEvent(event.toDTO())
+            true
+        } catch (e: Exception) {
+            false
+        }
```

Comment:
Here, you're potentially catching CancellationExceptions (same in all functions below)

Reply:
OK, I will improve the catches.